modelzoo/DLRM/train.py

```diff
@@ -50,107 +50,60 @@
-        all_columns = collections.OrderedDict(zip(column_headers, columns))
-        labels = all_columns.pop(LABEL_COLUMN[0])
-        features = all_columns
-        return features, labels
-
-    # Extract lines from input files using the Dataset API.
-    dataset = tf.data.TextLineDataset(filename)
-    dataset = dataset.shuffle(buffer_size=20000,
-                              seed=2021)  # set seed for reproducing
-    dataset = dataset.repeat(num_epochs)
-    dataset = dataset.prefetch(batch_size)
-    dataset = dataset.batch(batch_size)
-    dataset = dataset.map(parse_csv, num_parallel_calls=28)
-    dataset = dataset.prefetch(1)
-    return dataset
-
-
-def build_feature_cols():
-    dense_column = []
-    sparse_column = []
-    for column_name in FEATURE_COLUMNS:
-        if column_name in CATEGORICAL_COLUMNS:
-            categorical_column = tf.feature_column.categorical_column_with_hash_bucket(
-                column_name,
-                # hash_bucket_size=HASH_BUCKET_SIZES[column_name],
-                hash_bucket_size=10000,
-                dtype=tf.string)
-
-            sparse_column.append(
-                tf.feature_column.embedding_column(categorical_column,
-                                                   dimension=16,
-                                                   combiner='mean'))
-        else:
-            column = tf.feature_column.numeric_column(column_name, shape=(1, ))
-            dense_column.append(column)
-
-    return dense_column, sparse_column
-
-
 class DLRM():
     def __init__(self,
                  dense_column=None,
                  sparse_column=None,
                  mlp_bot=[512, 256, 64, 16],
                  mlp_top=[512, 256],
+                 optimizer_type='adam',
                  learning_rate=0.1,
                  inputs=None,
                  interaction_op='dot',
                  bf16=False,
+                 stock_tf=None,
+                 adaptive_emb=False,
                  input_layer_partitioner=None,
                  dense_layer_partitioner=None):
         if not inputs:
             raise ValueError('Dataset is not defined.')
-        self.dense_column = dense_column
-        self.sparse_column = sparse_column
+        self._feature = inputs[0]
+        self._label = inputs[1]
+
         if not dense_column or not sparse_column:
             raise ValueError('Dense column or sparse column is not defined.')
-        self.mlp_bot = mlp_bot
-        self.mlp_top = mlp_top
-        self.learning_rate = learning_rate
-        self.input_layer_partitioner = input_layer_partitioner
-        self.dense_layer_partitioner = dense_layer_partitioner
-
-        self.feature = inputs[0]
-        self.label = inputs[1]
-        self.bf16 = bf16
-        self._is_training = True
+        self._dense_column = dense_column
+        self._sparse_column = sparse_column
 
+        self.tf = stock_tf
+        self.bf16 = False if self.tf else bf16
+        self._is_training = True
```

Comment:
self._is_training to self.is_training. you will get a error here when you run this script.